sources/common/toddl.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"strconv"
+	"unicode"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+type BaseToDdl interface {
+	ToSpannerType(conv *internal.Conv, columnType schema.Type) (ddl.Type, []internal.SchemaIssue)
+}
+
+// TODO: refactor this file to avoid the duplication with postgres/toddl.go.
+// The core difference between the two files is toSpannerType, which maps
+// type ids (which differ between MySQL and PostgreSQL) to Spanner types.
+
+// schemaToDDL performs schema conversion from the source DB schema to
+// Spanner. It uses the source schema in conv.SrcSchema, and writes
+// the Spanner schema to conv.SpSchema.
+func SchemaToSpannerDDL(conv *internal.Conv, toddlImpl BaseToDdl) error {
+	// Tracks Spanner names that have been used for foreign key constraints
+	// and indexes. We use this to ensure we generate unique names when
+	// we map from Source to Spanner since Spanner requires all foreign
+	// key and index names to be distinct (you can't use the same name
+	// for a foreign key constraint and an index).
+	usedNames := make(map[string]bool)
+	// As Spanner uses same namespace for table names, foreign key constraint
+	// names and index names, we need to pre-populate usedNames with Spanner table
+	// names to handle collision with foreign key names and index names.
+	for _, srcTable := range conv.SrcSchema {
+		spTableName, err := internal.GetSpannerTable(conv, srcTable.Name)
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Couldn't map source table %s to Spanner: %s", srcTable.Name, err))
+			continue
+		}
+		usedNames[spTableName] = true
+	}
```

Comment:
I think this is the older version of the code. In particular this drops the changes from https://github.com/cloudspannerecosystem/harbourbridge/pull/168.

Reply:
removed